common-service-core/src/main/java/org/zowe/apiml/services/BasicInfoService.java

```diff
@@ -0,0 +1,159 @@
+import static org.zowe.apiml.services.ServiceInfoUtils.getVersion;
+
+/**
+ * Similar to {@link org.zowe.apiml.gateway.services.ServicesInfoService} service which does not depend on gateway-service components.
+ * Following properties left blank:
+ * {@link ServiceInfo.Service#homePageUrl} and {@link ServiceInfo.ApiInfoExtended#swaggerUrl}
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class BasicInfoService {
+
+    private final EurekaClient eurekaClient;
+    private final EurekaMetadataParser eurekaMetadataParser;
+
+    public List<ServiceInfo> getServicesInfo() {
+        return eurekaClient.getApplications().getRegisteredApplications()
+                .stream().map(this::getServiceInfo)
+                .collect(Collectors.toCollection(LinkedList::new));
+    }
+
+    private ServiceInfo getServiceInfo(Application application) {
+        String serviceId = application.getName().toLowerCase();
+
+        List<InstanceInfo> appInstances = application.getInstances();
+        if (ObjectUtils.isEmpty(appInstances)) {
+            return ServiceInfo.builder()
+                    .serviceId(serviceId)
+                    .status(InstanceInfo.InstanceStatus.DOWN)
+                    .build();
+        }
+
+        return ServiceInfo.builder()
+                .serviceId(serviceId)
+                .status(getStatus(appInstances))
+                .apiml(getApiml(appInstances))
+                .instances(getInstances(appInstances))
+                .build();
+    }
+
+    /**
+     * uses simplified:
+     * - getApiInfos
+     * - getApiInfos
+     */
+    private ServiceInfo.Apiml getApiml(List<InstanceInfo> appInstances) {
+        return ServiceInfo.Apiml.builder()
+                .apiInfo(getApiInfos(appInstances))
+                .service(getService(appInstances))
+                .authentication(getAuthentication(appInstances))
+                .build();
+    }
+
+
+    /**
+     * simplified version, following part is excluded:
+     * - homePageUrl
+     */
+    private ServiceInfo.Service getService(List<InstanceInfo> appInstances) {
+        InstanceInfo instanceInfo = getInstanceWithHighestVersion(appInstances);
+
+        return ServiceInfo.Service.builder()
+                .title(instanceInfo.getMetadata().get(SERVICE_TITLE))
+                .description(instanceInfo.getMetadata().get(SERVICE_DESCRIPTION))
+                .build();
+    }
+
+    /**
+     * Simplified version, following properties are excluded:
+     * - baseUrl
+     * - swaggerUrl
+     */
+    private List<ServiceInfo.ApiInfoExtended> getApiInfos(List<InstanceInfo> appInstances) {
+        return appInstances.stream()
+                .map(instanceInfo -> new AbstractMap.SimpleEntry<>(instanceInfo, eurekaMetadataParser.parseApiInfo(instanceInfo.getMetadata())))
+                .flatMap(entry -> entry.getValue().stream()
+                        .map(apiInfo -> ServiceInfo.ApiInfoExtended.builder()
```

Comment:
Could you please solve Sonar issue here? see https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=3051&id=zowe_api-layer&open=AYpQssMjW7YGbxMnQVBn

Reply:
there are 2 builders in base and sub classes. Sonar do not see it.